gravitee-am-management-api/gravitee-am-management-api-service/src/test/java/io/gravitee/am/management/service/impl/upgrades/NonBCryptIterationsRoundsUpgraderTest.java

```diff
@@ -0,0 +1,123 @@
+
+    @Mock
+    private SystemTaskRepository systemTaskRepository;
+
+    @InjectMocks
+    private NonBCryptIterationsRoundsUpgrader upgrader;
+
+    @Before
+    public void setUp() throws Exception {
+        SystemTask task = new SystemTask();
+        task.setOperationId("op1");
+        Mockito.when(systemTaskRepository.updateIf(Mockito.any(), Mockito.any()))
+                .thenReturn(Single.just(task));
+    }
+
+    @Test
+    public void shouldUpdateAllNonBCryptIDPPasswordEncoderOptions() throws Exception {
+        // given
+        Single<IdentityProvider> updateResult = Single.just(new IdentityProvider());
+        ArgumentCaptor<UpdateIdentityProvider> captor = ArgumentCaptor.captor();
+        Mockito.when(idpService.findAll())
+                .thenReturn(just(
+                        idp("1", "{\"uri\":\"mongodb://localhost:27017\",\"host\":\"localhost\",\"port\":27017,\"enableCredentials\":false,\"passwordCredentials\":null,\"database\":\"m2\",\"usersCollection\":\"users\",\"findUserByUsernameQuery\":\"{username: ?}\",\"findUserByEmailQuery\":\"{email: ?}\",\"usernameField\":\"username\",\"passwordField\":\"password\",\"passwordEncoder\":\"SHA-256+MD5\",\"passwordEncoderOptions\":{\"rounds\":106},\"useDedicatedSalt\":false,\"passwordSaltLength\":32,\"passwordSaltFormat\":\"DIGEST\",\"userProvider\":true,\"usernameCaseSensitive\":false}"),
+                        idp("2", "{\"uri\":\"mongodb://localhost:27017/?connectTimeoutMS=1000&socketTimeoutMS=1000\",\"host\":\"localhost\",\"port\":\"27017\",\"enableCredentials\":false,\"database\":\"gravitee-am\",\"usersCollection\":\"idp_users_58b8632e-a482-4c11-b863-2ea482cc1169\",\"findUserByUsernameQuery\":\"{username: ?}\",\"findUserByEmailQuery\":\"{email: ?}\",\"usernameField\":\"username\",\"passwordField\":\"password\",\"passwordEncoder\":\"BCrypt\",\"passwordEncoderOptions\":{\"rounds\":10}}"),
+                        idp("3", "{\"uri\":\"mongodb://localhost:27017/?connectTimeoutMS=1000&socketTimeoutMS=1000\",\"host\":\"localhost\",\"port\":\"27017\",\"enableCredentials\":false,\"database\":\"gravitee-am\",\"usersCollection\":\"idp_users_58b8632e-a482-4c11-b863-2ea482cc1169\",\"findUserByUsernameQuery\":\"{username: ?}\",\"findUserByEmailQuery\":\"{email: ?}\",\"usernameField\":\"username\",\"passwordField\":\"password\",\"passwordEncoder\":\"BCrypt\",\"passwordEncoderOptions\":{\"rounds\":100}}")
+                ));
+
+
+        Mockito.when(idpService.update(Mockito.any(), Mockito.any(), Mockito.any(), captor.capture(), Mockito.any(), Mockito.anyBoolean()))
+                .thenReturn(updateResult);
+
+        // when
+        TestObserver<Boolean> observer = new TestObserver();
+        upgrader.processUpgrade("op1", new SystemTask(), "op11")
+                .subscribe(observer);
+
+        observer.assertNoErrors();
+        updateResult.test().assertValueCount(1);
+
+        Assertions.assertEquals(1, captor.getAllValues().size());
```

Comment:
it is size 1 but from findAll() we have 3 elements. Can we have a test to verify that multiple elements have been updated?